Userland/Libraries/LibGfx/JPEGLoader.cpp

```diff
@@ -1099,6 +1099,55 @@ static ErrorOr<void> compose_bitmap(JPEGLoadingContext& context, Vector<Macroblo
     return {};
 }
 
+static bool is_app_marker(Marker const marker)
+{
+    return marker >= JPEG_APPN0 && marker <= JPEG_APPNF;
+}
+
+static bool is_miscellaneous_or_table_marker(Marker const marker)
+{
+    // B.6 - Summary
+    // See: Figure B.17 – Flow of marker segment
+
+    bool const is_misc = marker == JPEG_COM || marker == JPEG_DRI || is_app_marker(marker);
+    bool const is_table = marker == JPEG_DQT || marker == JPEG_DAC || marker == JPEG_DHT;
+
+    return is_misc || is_table;
+}
+
+static ErrorOr<void> handle_miscellaneous_or_table(AK::SeekableStream& stream, JPEGLoadingContext& context, Marker const marker)
+{
+    if (is_app_marker(marker)) {
+        TRY(read_app_marker(stream, context, marker - JPEG_APPN0));
+        return {};
+    }
+
+    switch (marker) {
+    case JPEG_COM:
```

Comment:
nit: COM is just a comment. Not sure it's worth printing a TODO for that – we can just skip it without a comment, maybe?

(you could switch the case branches, put the dbgln_if only after the DAC one and then `[[fallthrough]]` to the COM one)

Reply:
I don't even know if comments are used by some applications, but anyway, I'm not really fond of pretending we are handling a segment if we aren't.

Reply:
I suppose that handling them would be to report them to the user in something like ImageViewer->Image Properties.
I know that it's irrelevant to the image data, but JPEGLoader still have the responsibility to expose metadata to the user.
We do log that we are not supporting it when we encounter an APP segment.

`dbgln` isn't the standard output, so I don't think it applies here.

Anyway, it replicates the old behavior and I won't try to sneak some behavior changes in this commit. We can still fix that later.